codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -496,10 +493,10 @@ private void readPriorityFrame(ChannelHandlerContext ctx, ByteBuf payload,
         long word1 = payload.readUnsignedInt();
         boolean exclusive = (word1 & 0x80000000L) != 0;
         int streamDependency = (int) (word1 & 0x7FFFFFFFL);
+        short weight = (short) (payload.readUnsignedByte() + 1);
         if (streamDependency == streamId) {
             throw streamError(streamId, PROTOCOL_ERROR, "A stream cannot depend on itself.");
         }
-        short weight = (short) (payload.readUnsignedByte() + 1);
```

Comment:
fwiw, now that you've sliced the buffer I don't think you need to do this re-ordering.